File: backend-web-api/Models/Transaction.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;
using backend_web_api.Enums;

namespace backend_web_api.Models
{
    public class Transaction
    {
        public int Id { get; set; }
        [MaxLength(400)] public string Description { get; set; }
        public decimal Amount { get; set; }
        public TransactionType Type { get; set; }
        public int PersonId { get; set; }
        public int CategoryId { get; set; }
        [JsonIgnore] public Person Person { get; set; }
        [JsonIgnore] public Category Category { get; set; }
    }
}

File: frontend/src/types/Transaction.ts
export const TransactionType = {
  Expense: "Expense",
  Income: "Income"
} as const;

export type TransactionType = typeof TransactionType[keyof typeof TransactionType];

export interface Transaction {
  id: number;
  description: string;
  amount: number;
  type: TransactionType;
  personId: number;
  categoryId: number;
}

export const transactionLabels: Record<TransactionType, string> = {
  Expense: "Despesa",
  Income: "Receita"
}